core-bundle/src/Resources/contao/elements/ContentMedia.php

```diff
@@ -130,22 +130,27 @@ protected function compile()
 		// Convert the language to a locale (see #5678)
 		$strLanguage = LocaleUtil::formatAsLocale($objPage->language);
 
+		// Determine caption
+		$strCaption = $this->playerCaption;
+
 		// Pass File objects to the template
 		while ($objFiles->next())
 		{
-			$arrMeta = StringUtil::deserialize($objFiles->meta);
+			$objMeta = $objFiles->current()->getMetadata($strLanguage);
+			$strTitle = null;
 
-			if (\is_array($arrMeta) && isset($arrMeta[$strLanguage]))
-			{
-				$strTitle = $arrMeta[$strLanguage]['title'];
-			}
-			else
+			if (null !== $objMeta)
 			{
-				$strTitle = $objFiles->name;
+				$strTitle = $objMeta->getTitle();
+
+				if (empty($strCaption))
```

Comment:
This check should be first in the loop, so we don't call `getMetadata()` etc. if we already know that `$strCaption` will not be overwritten anyway.

Reply:
That won't work, because we always need the title.

Reply:
> How can this be correct? It seems pretty random to me.

It is essentially random as you cannot sort the sources in the back end. But since there can only be one caption we can only extract the caption from one file - which might as well just be whichever source happens to be first.

> IIRC, the caption of the file manager will overwrite the caption set in the player element, won't it?

No, not in the current implementation. If a caption is set in the player element, `if (empty($strCaption))` will be `false` and thus no caption is used from any source.

Reply:
Exactly. 😄

Reply:
? The caption of the player element will overwrite the caption of the file manager, not the other way around is what I am saying ;)